lms/services/http.py

```diff
@@ -93,9 +102,18 @@ def request(
 
             The invalid response will be available as
             HTTPValidationError.response.
+
+        :raise NoOAuth2Token: If oauth=True was given but we don't have an
+            access token for the current user in our DB
         """
         response = None
 
+        if oauth:
+            kwargs.setdefault("headers", {})
+            assert "Authorization" not in kwargs["headers"]
```

Comment:
Some python programmers might be triggered by this assert and prefer something like 

```
if "Authorization" in kwargs["headers"]:
   raise ValueError("Can't use custom auth and Oauth simultaneously")
```
I have a preference for something like that, but only a small one.